service/common/src/main/java/org/apache/polaris/service/auth/JWTBroker.java

```diff
@@ -62,29 +65,42 @@ public abstract class JWTBroker implements TokenBroker {
 
   @Override
   public DecodedToken verify(String token) {
+
+    DecodedJWT decodedJWT;
+    try {
+      decodedJWT = JWT.decode(token);
+    } catch (JWTDecodeException e) {
+      LOGGER.error("Failed to decode the token", e);
+      throw new NotAuthorizedException("Failed to decode the token");
+    }
+
+    if (!decodedJWT.getIssuer().equals(ISSUER_KEY)) {
+      throw new UnprocessableEntityException("Unexpected issuer: %s", decodedJWT.getIssuer());
```

Comment:
Here, we need a way to signal the caller that the token issuer is wrong. This must be a different error so that the caller can distinguish this situation from a validation error situation. 

I picked `UnprocessableEntityException` out of better choices.